internal/secrets/list/list.go

```diff
@@ -38,7 +38,11 @@ func Run() error {
 			return err
 		}
 
-		req, err := http.NewRequest("GET", "https://api.supabase.io/v1/projects/"+projectRef+"/secrets", nil)
+		supabaseAPI := os.Getenv("SUPABASE_API_URI")
+		if supabaseAPI == "" {
+			supabaseAPI = "https://api.supabase.io"
+		}
+		req, err := http.NewRequest("GET", supabaseAPI+"/v1/projects/"+projectRef+"/secrets", nil)
```

Comment:
Instead of introducing a new env var, shall we use [gock](https://github.com/h2non/gock) to mock out any supabase api requests for now? Eventually we want to expose `SUPABASE_HOST` properly as a `viper` config.

Reply:
We agreed on using gock in unit tests. 

Since this extremely unlikely to break anything, i think it is ok as temp solution, and can be reworked in future (because not everything should be in one pr right).

Reply:
Well, I'm hoping to reduce the scope of this PR by reverting this env var.

Especially since we have no control over users who set this env var once it's released. It's a small problem given the number of cli users right now. But for the peace of mind, I'd rather not introduce new config that's used for testing only. 

When we are ready to expose it for changing the supabase host that cli talks to, we can look into it again. That way we can also plan its naming convention more carefully. What do you think?

Reply:
I don't think this is such a problem, but we may change env var to be sure noone will override it. 
To `SUPABASE_CLI_SUPABASE_API_URI` for example. With this I doubt any of our users accidentally override it. 

This is good for us since we actually will be able to test some new things on staging also.

Reply:
> This is good for us since we actually will be able to test some new things on staging also.

Yup, I agree it is good for us.

@soedirgo suggests `SUPABASE_INTERNAL_API_URL` to be more concise. I'm ok with that also.

Reply:
I don't mind having it as an env, as long as we make it clear that it's for internal use only, e.g. by renaming it to `_SUPABASE_ADMIN_API_HOST` or `SUPABASE_INTERNAL_ADMIN_API_HOST`.

btw we started off using `viper`, but moved off of it - iirc something to do with flexibility and fine-grained error messages.

Reply:
> btw we started off using viper, but moved off of it - iirc something to do with flexibility and fine-grained error messages.

Hmm I would be interested to read the discussion thread ^. The way `viper` standardises precedence order sounds too good to pass up.

Reply:
`SUPABASE_INTERNAL_API_HOST` 👍

Reply:
I will apply all suggested changes later today

Reply:
Err, `s/we started off/I started off/` 😄. We still use it here: https://github.com/supabase/cli/blob/9c4b8a9801a50a553fb08e5b79180999482913c9/internal/utils/config.go#L289-L307

These are for `supabase/config.{json,toml}` though which is associated w/ the project and not the CLI - these would be different once we have some sort of global config (i.e. the ones in `~/.supabase`, not `./supabase`

Reply:
also we can prob remove this deprecated config handling code now - will do it in another PR